sdk/iot/hub/inc/az_iot_hub_client.h

```diff
@@ -182,15 +182,16 @@ typedef struct az_iot_hub_client_properties
   struct
   {
     az_span properties;
+    int32_t properties_length;

```

Comment:
@danewalton, @CIPop - please review this change.

Reply:
This seems reasonable to me. If we end up not liking it we can change it later.

Reply:
Nit: Name is a bit confusing (it's the capacity of the buffer). @danewalton I think it may be clearer now to create our own buffer/len/capacity for properties and return az_spans when needed. 

@ahsonkhan I agree with @danewalton I think this solution is good for now.

Reply:
> Nit: Name is a bit confusing (it's the capacity of the buffer). 

`properties_length` isn't the capacity of the buffer but rather how much data has been written into `properties` so far.

What name would you suggest?

Reply:
Oh, in that case it is very confusing... We could have `properties_buffer` and `properties_written` or something on those lines.